src/modules/UI/selectors.js

```diff
@@ -112,3 +112,29 @@ export const getSceneState = (state: State, sceneKey: string) => {
   const sceneState = getScenesState(state)[sceneKey]
   return sceneState
 }
+
+export const getExchangeRate = (state: State, fromCurrencyCode: string, toCurrencyCode: string): number => {
+  const fromCurrencyCleaned = fromCurrencyCode.replace('iso:', '')
+  const toCurrencyCleaned = toCurrencyCode.replace('iso:', '')
+  const exchangeRates = state.exchangeRates
+  const rateKey = `${fromCurrencyCleaned}_${toCurrencyCleaned}`
+  const rate = exchangeRates[rateKey] ? exchangeRates[rateKey] : 0
+  // if (fromCurrencyCleaned === toCurrencyCleaned) rate = 1 // handle case that occurs while app is loading
+  return rate
+}
+
+export const convertCurrency = (state: State, fromCurrencyCode: string, toCurrencyCode: string, amount: number = 1) => {
+  const exchangeRate = getExchangeRate(state, fromCurrencyCode, toCurrencyCode)
+  const convertedAmount = amount * exchangeRate
+  return convertedAmount
+}
+
+export const convertCurrencyFromExchangeRates = (exchangeRates: { [string]: number }, fromCurrencyCode: string, toCurrencyCode: string, amount: number) => {
+  const fromCurrencyCleaned = fromCurrencyCode.replace('iso:', '')
```

Comment:
Convert a currency **with the Redux exchange rates as a parameter**. Since this function does not access redux state that means it can be called from the component rather than the connector

Reply:
Only grab one exchange rate

Reply:
^ that is what `convertCurrency` does